File: dev-docs/snippets/documents/create-with-alias.swift
import JsBaoClient

// Create a document and assign an alias atomically with typed
// `CreateWithAliasOptions`. Returns a `CreateWithAliasResult` (incl. the id).
func createWithAlias(client: JsBaoClient) async throws {
  // #region example
  let result = try await client.documents.createWithAlias(
    options: CreateWithAliasOptions(
      title: "User Notes",
      alias: AliasRef(scope: .user, aliasKey: "notes")
    )
  )
  // #endregion example
  _ = result.documentId
}

File: dev-docs/snippets/documents/create-with-alias.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Create a document and assign an alias atomically; fails if the alias exists.
export async function createWithAlias(client: JsBaoClient) {
  // #region example
  const result = await client.documents.createWithAlias({
    title: "User Notes",
    alias: { scope: "user", aliasKey: "notes" },
  });
  // #endregion example
  return result;
}
